server/server.go

```diff
@@ -279,11 +265,23 @@ func (s *Server) Run(host, token string, insecure bool) error {
 	}
 	s.k8s = k
 	s.iam = iam.NewClient(s.BaseRoleARN)
-	model := store.NewStore(s.IAMRoleKey, s.DefaultIAMRole, s.NamespaceRestriction, s.NamespaceKey, s.iam)
-	s.store = model
-	s.k8s.WatchForPods(kube2iam.NewPodHandler(model))
-	s.k8s.WatchForNamespaces(kube2iam.NewNamespaceHandler(model))
+	s.roleProcessor = processor.NewRoleProcessor(s.IAMRoleKey, s.DefaultIAMRole, s.NamespaceRestriction, s.NamespaceKey, s.iam, s.k8s)
+	podSynched := s.k8s.WatchForPods(k8s.NewPodHandler(s.IAMRoleKey))
+	namespaceSynched := s.k8s.WatchForNamespaces(k8s.NewNamespaceHandler(s.NamespaceKey))
+
+	synced := false
+	for i := 0; i < defaultCacheSyncAttempts && !synced; i++ {
+		synced = cache.WaitForCacheSync(nil, podSynched, namespaceSynched)
```

Comment:
It was more around ensuring that both the pods and namespaces were indexed to ensure that you couldn't "sneak in" a role.  This only happens @ startup time, so there is the possibility that you could get the default role immediately if the new pod hasn't been indexed.  

What this PR aims to solve is a permanent caching of those incorrect roles.